rm-datasource/src/main/java/io/seata/rm/datasource/DataCompareUtils.java

```diff
@@ -65,6 +65,9 @@ public static boolean isFieldEquals(Field f0, Field f1) {
                             if (f1Type == Types.TIMESTAMP && f1.getValue().getClass().equals(String.class)) {
                                 f1.setValue(Timestamp.valueOf(f1.getValue().toString()));
                             }
+                            if (!checkType(f0.getValue())) {
+                                return true;
```

Comment:
return false?

Reply:
ignore other type compare

Reply:
> @xingfudeshi ignore other type compare, because other type not override equals and hashcode, when comparing f0.value and f1.value, the actual comparison is the object address. the compare result certain false.

If true is returned here, does f0. value equals to f1. value?

Reply:
if f0.value is not any instance of there types,then f0.value equals f1.value?

Reply:
what i mean is if the other type dose not override equals and hashcode method,so we don't need to compare them,right?

Reply:
you are right, I understand your meanning, I will update it later.